### PR TITLE
allow playlist in NDEF url

### PR DESCRIPTION
**Pull request by luka6000**
This modification will allow to read Spotify and Apple Music playlist url from NDEF tag and use it in HA incoming as event music_tag. Additionally, there is a write_music_tag service which allow to write any url as NDEF on the tag

### CODE REVIEW

**Comment by adonno** [comment]
@jesserockz mind having a look please ?

**Comment by luka6000** [comment]
sorry guy, I've completely refactored the code to parse the tag data only once. Now it's more clean and works as expected.

File: tagreader.yaml [hunk 1]
```diff
@@ -222,7 +249,90 @@ pn532_i2c:
...
+        else:
+        - if:
+            condition:
+              lambda: 'return ( id(source)=="hass" );'
+            then:
+            - homeassistant.tag_scanned: !lambda 'return id(info);'
+            else:
+            - homeassistant.event:
+                event: esphome.music_tag
+                data_template:
```
Comment (jesserockz) [inline tagreader.yaml:320]: This doesnt need to be `data_template` (Yes, The documentation is a little off here)
```suggestion
                data:
```

**Review by jesserockz — COMMENTED** [review]
Changes overall look good. Just depends if you want to include them in the main repo yaml @adonno ?

**Comment by adonno** [comment]
> Changes overall look good. Just depends if you want to include them in the main repo YAML @adonno ?

@jesserockz I thought that could be useful since a lot of people want to use it as a jukebox and would you advise against it ? 
if so we could create a second YAML for jukebox only

**Comment by jesserockz** [comment]
@adonno No I have nothing against adding this into the main YAML. I do think it would make it easier for people to then use unmodified.

**Review by adonno — APPROVED** [review]

File: tagreader.yaml [hunk 2]
```diff
@@ -190,6 +213,48 @@ pn532_i2c:
...
+    - if:
+        condition:
+          lambda: |
+            if (tag.has_ndef_message()) {
+              auto message = tag.get_ndef_message();
+              auto records = message->get_records();
+              for (auto &record : records) {
+                std::string payload = record->get_payload();
+                std::string type = record->get_type();
+                size_t applemusic = payload.find("https://open.spotify.com");
```
[the remark this answers is not in the document]
Comment (luka6000) [reply tagreader.yaml]: yes, I changed it with later commits

File: tagreader.yaml [hunk 3]
```diff
@@ -213,6 +213,48 @@ pn532_i2c:
...
+                size_t spotify = payload.find("https://music.apple.com");
```
[the remark this answers is not in the document]
Comment (luka6000) [reply tagreader.yaml]: yes, I changed it with later commits

**Pull request merged**